mlir/include/mlir/Dialect/MIOpen/XdlopsCodeSelection.h

```diff
@@ -436,14 +433,31 @@ struct XdlopsCodeSelection {
         imms.push_back({0, 0, 0});
         argType = VectorType::get({2}, b.getBF16Type());
       } else {
-        llvm::errs() << "Unsupported case:\n";
-        // llvm::errs() << "M, N, K:" << M << " " << N << " " << K << "\n";
-        llvm::errs() << "MPerWave: " << MPerWave << "\n";
-        llvm::errs() << "NPerWave: " << NPerWave << "\n";
-        llvm::errs() << "dataType: ";
-        dataType.dump();
-        llvm::errs() << "\n";
-        llvm_unreachable("Can't meaningfully continue xdlop generation");
+        dumpUnsupported(dataType, MPerWave, NPerWave);
+      }
+    } else if (dataType == b.getIntegerType(8)) {
+      if (MPerWave == 32 && NPerWave == 32) {
+        mfmaInstr = "mfma_i32_32x32x8i8";
+        MPerXdlops = 32;
+        NPerXdlops = 32;
+        MRepeats = 1;
+        NRepeats = 1;
+        vectorType = VectorType::get({16}, b.getI32Type());
+        vectorNumber = 1;
+        imms.push_back({0, 0, 0});
+        argType = b.getIntegerType(8);
```

Comment:
Is `argType` here a vector of i8 or just a single i8?

Reply:
Nice catch, this should be vector<4xi8> lookin at how it is defined for other types. Will amend.

Reply:
Addressed in d1a2fab